File: src/test/java/TestExampleFiles.java
import model.Node;
import model.RuleUsageCounter;
import org.junit.Test;
import parser.TestDataParser;
import parser.TreeParser;

import java.util.List;

import static org.junit.Assert.fail;

public class TestExampleFiles {

    @Test
    public void compareExampleFilesForTrace() {
        TreeParser inputParser = new TreeParser();
        List<Node> input = inputParser.parseTreeFile("wsj_0100.tree");
        input.forEach(Node::removeTraceTrees);

        TreeParser testDataParser = new TreeParser();
        List<Node> testData = testDataParser.parseTreeFile("wsj_0100.noTrace");

        for (int x = 0; x < input.size(); x++) {
            if (!input.get(x).toString().equals(testData.get(x).toString())) {
                fail();
            }
        }
    }

    @Test
    public void compareExampleFilesForGrammar() {
        TreeParser inputParser = new TreeParser();
        List<Node> input = inputParser.parseTreeFile("wsj_0100.noTrace");

        TestDataParser testParser = new TestDataParser();
        List<String> grammar = testParser.readGrammar();

        RuleUsageCounter counter = new RuleUsageCounter(input);
        List<String> inputGrammar = counter.asStringList();

        inputGrammar.forEach(foundRule -> {
            if (!grammar.contains(foundRule)) {
                fail();
            }
        });
    }
}

File: src/test/kotlin/TestExampleFiles.kt
import model.RegularRuleCollector
import model.RuleUsageCounter
import model.WordUsageCounter
import org.junit.Assert.assertEquals
import org.junit.Assert.fail
import org.junit.Test
import parser.TestDataParser
import parser.TreeParser

class TestExampleFiles {

    @Test
    fun compareExampleTraceFile() {
        val inputParser = TreeParser()
        val input = inputParser.parseTreeFile("wsj_0100.tree")
        input!!.forEach { it.removeTraceTrees() }

        val testDataParser = TreeParser()
        val testData = testDataParser.parseTreeFile("wsj_0100.noTrace")

        input.indices
                .filter { input[it].toString() != testData!![it].toString() }
                .forEach { fail() }
    }

    @Test
    fun compareExampleGrammarFile() {
        val inputParser = TreeParser()
        val input = inputParser.parseTreeFile("wsj_0100.noTrace")

        val testParser = TestDataParser()
        val grammar = testParser.readGrammar()

        val counter = RuleUsageCounter(input!!)
        val inputGrammar = counter.asStringList()

        inputGrammar.forEach { foundRule ->
            if (!grammar.contains(foundRule)) {
                fail()
            }
        }
    }

    @Test
    fun compareExampleLexiconFile() {
        val inputParser = TreeParser()
        val input = inputParser.parseTreeFile("wsj_0100.noTrace")

        val testParser = TestDataParser()
        val grammar = testParser.readLexicon()

        val counter = WordUsageCounter(input!!)
        val inputGrammar = counter.asStringList()

        assertEquals(grammar.size.toLong(), inputGrammar.size.toLong())
    }

    @Test
    fun compareRegularRulesToSolution() {
        val inputParser = TreeParser()
        val input = inputParser.parseTreeFile("wsj_0100.noTrace")

        val testParser = TestDataParser()
        val regularRulesSolution = testParser.readRegularRules()

        val regularRules = RegularRuleCollector(input!!)

        assertEquals(regularRules.size, regularRulesSolution.size)
    }

}
